apis/workstations/v1beta1/workstation_identity.go

```diff
@@ -0,0 +1,141 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1beta1
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/k8s-config-connector/apis/common"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// WorkstationIdentity defines the resource reference to Workstation.
+type WorkstationIdentity struct {
+	parent *WorkstationParent
+	id     string
+}
+
+func (i *WorkstationIdentity) String() string {
+	return i.parent.String() + "/workstations/" + i.id
+}
+
+func (i *WorkstationIdentity) ID() string {
+	return i.id
+}
+
+func (i *WorkstationIdentity) Parent() *WorkstationParent {
+	return i.parent
+}
+
+type WorkstationParent struct {
```

Comment:
Or we can use `WorkstationClusterIdentity`.  But we need some code refactoring to allow (and only allow) the `WorkstationClusterIdentity` initialization to be done via `NewWorkstationClusterIdentity` or its child's `NewIdentity` function.